specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/Sessions.json

```diff
@@ -0,0 +1,374 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ContainerApps API Client",
+    "version": "2024-02-02-preview"
+  },
+  "schemes": [
+    "https"
+  ],
+  "host": "management.azure.com",
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/sessionPools/{sessionPoolName}/generateSessions": {
+      "post": {
+        "tags": [
+          "ContainerAppsSessionPoolSessions"
+        ],
+        "summary": "Generate Sessions from a Session Pool.",
+        "operationId": "ContainerAppsSessionPoolSessionsGeneration_ListGenerated",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "sessionPoolName",
+            "in": "path",
+            "description": "Name of the Session Pool.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z][a-z0-9]*$",
+            "minLength": 3,
+            "maxLength": 63
+          },
+          {
+            "name": "generateSessionsEnvelope",
+            "in": "body",
+            "description": "Properties used to generate a session",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GenerateSessionsRequest"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
if this is async operation then status should be 202  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md

And if it is not an async operation, then it needs to be under <1 sec completion SLA I guess.

Reply:
@yitaopan I am assuming server implementation is also as per API spec as previously it was mentioned as sync operation in API spec.